contrib/src/main/java/com/datatorrent/contrib/parser/JsonParser.java

```diff
@@ -0,0 +1,241 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonschema.exceptions.ProcessingException;
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+import com.github.fge.jsonschema.report.ProcessingMessage;
+import com.github.fge.jsonschema.report.ProcessingReport;
+import com.github.fge.jsonschema.util.JsonLoader;
+import com.google.common.annotations.VisibleForTesting;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.lib.parser.Parser;
+import com.datatorrent.lib.util.KeyValPair;
+import com.datatorrent.netlet.util.DTThrowable;
+
+/**
+ * Operator that parses a json string tuple against a specified json schema and
+ * emits JSONObject on one port POJO on other port and tuples that could not be
+ * parsed on error port.<br>
+ * Schema is specified in a json format as per http://json-schema.org/ <br>
+ * Example for the schema can be seen here http://json-schema.org/example1.html <br>
+ * User can choose to skip validations by not specifying the schema at all. <br>
+ * <br>
+ * <b>Properties</b> <br>
+ * <b>jsonSchema</b>:schema as a string<br>
+ * <b>clazz</b>:Pojo class <br>
+ * <b>Ports</b> <br>
+ * <b>in</b>:input tuple as a String. Each tuple represents a json string<br>
+ * <b>parsedOutput</b>:tuples that are validated against the schema are emitted
+ * as JSONObject on this port<br>
+ * <b>out</b>:tuples that are validated against the schema are emitted as pojo
+ * on this port<br>
+ * <b>err</b>:tuples that do not confine to schema are emitted on this port as
+ * KeyValPair<String,String><br>
+ * Key being the tuple and Val being the reason.
+ * 
+ * 
+ * @displayName JsonParser
+ * @category Parsers
+ * @tags json pojo parser
+ * @since 3.2.0
+ */
+@InterfaceStability.Evolving
+public class JsonParser extends Parser<byte[], KeyValPair<String, String>>
+{
+
+  /**
+   * Contents of the schema.Schema is specified as per http://json-schema.org/
+   */
+  private String jsonSchema;
+  private transient JsonSchema schema;
+  private transient ObjectMapper objMapper;
+  /**
+   * output port to emit validate records as JSONObject
+   */
+  public transient DefaultOutputPort<JSONObject> parsedOutput = new DefaultOutputPort<JSONObject>();
+  /**
+   * metric to keep count of number of tuples emitted on {@link #parsedOutput}
+   * port
+   */
+  @AutoMetric
+  long parsedOutputCount;
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    parsedOutputCount = 0;
+  }
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    try {
+      if (jsonSchema != null) {
+        JsonSchemaFactory factory = JsonSchemaFactory.byDefault();
+        JsonNode schemaNode = JsonLoader.fromString(jsonSchema);
+        schema = factory.getJsonSchema(schemaNode);
+      }
+      objMapper = new ObjectMapper();
```

Comment:
By default object mapper will check if all fields specified in json are available in the class, if it does not find such mapping it will fail the parsing. you need to relax this condition by setting
```
mapper.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

This will allow users to specify pojo with less number of fields than what are present in json. This way user can only extract interested fields from json.